### PR TITLE
time-cutoff version 2.0

### PR DESCRIPTION
**Pull request by cfichtlscherer**
# Description

I'm very sorry to open a new pull-request for this, but something is wrong with the old one and I can't figure out what's wrong even after several hours (among other things, clang-format error is issued for lines I haven't edited).
The original pull-request can be found here: https://github.com/openmc-dev/openmc/pull/2454.

 I know this is not optimal, but I have added all the changes here again.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) on any C++ source files (if applicable)
- [x] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)

### CODE REVIEW

File: src/particle.cpp [hunk 1]
```diff
@@ -203,11 +210,25 @@ void Particle::event_advance()
   double distance = std::min(boundary().distance, collision_distance());
 
   // Advance particle in space and time
+  // Short-term solution until the surface source is revised and we can use
+  // this->move_distance(distance)
```
Comment (paulromano) [inline src/particle.cpp:214]: Why can't you use this?

**Review by paulromano — COMMENTED** [review]
@cfichtlscherer Many apologies for the delay in reviewing this! Just one comment/question:

File: src/particle.cpp (hunk 1 above)
Comment (cfichtlscherer) [reply src/particle.cpp:214]: No worries. 

This was due to the bug you/we found in the **surface source test**. 
see: https://github.com/openmc-dev/openmc/pull/2454#discussion_r1263308672

As we use this operation (moving the particle) at several positions in the code it makes sense to functionalize it.

I thought we add it like this and someone (maybe you, cause I have no idea what is going on there...) will make the changes for the surface source / the surface source test and then we use `move_distance` also here. So its an interm solution.

What do you think?

Comment (paulromano) [reply src/particle.cpp:214]: Ahh right, I totally forgot about the weirdness with that test. Thanks for reminding me :smile: I still don't have a good idea what's going on with it so, as you suggest, we'll just punt for now and leave the code as is.

**Review by paulromano — APPROVED** [review]

**Comment by cfichtlscherer** [comment]
Thanks a lot for reviewing! And nice to have this feature added.

**Pull request merged**